packages/graphile-build-pg/src/queryFromResolveDataFactory.js

```diff
@@ -310,10 +310,12 @@ exists(
         queryBuilder.whereBound(sql.fragment`false`, isAfter);
       }
       const orderByExpressionsAndDirections = queryBuilder.getOrderByExpressionsAndDirections();
-      if (
-        orderByExpressionsAndDirections.length > 0 &&
-        queryBuilder.isOrderUnique()
-      ) {
+      if (orderByExpressionsAndDirections.length > 0) {
+        if (!queryBuilder.isOrderUnique()) {
+          throw new Error(
+            "The cursor requires a primary key or an unique key combination for orderBy clause"
```

Comment:
```suggestion
            "The order supplied is not unique, so before/after cursors cannot be used. Please ensure the supplied order includes all the columns from the primary key or a unique constraint."
```

Reply:
Yes, I think it's clearer that way. I've just merged it. Thanks.